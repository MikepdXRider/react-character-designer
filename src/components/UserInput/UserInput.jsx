import React from 'react'
import Select from '../Select/Select.jsx'

export default function UserInput({setHead, head, setTorso, torso, setBottom, bottom, newCatchPhraseStr, setNewCatchphraseStr, addCatchphrase}) {
    return (
        <section>
            <label>
                Head
                <Select
                    optionsArr={['cowboy-hat', 'green-hat', 'top-hat', 'wizard-hat']}
                    setState={setHead}
                    state={head}
                />
            </label>
            <label>
                Torso
                <Select
                    optionsArr={['cowboy-torso', 'formal-torso', 'jacket-torso']}
                    setState={setTorso}
                    state={torso}
                />
            </label>
            <label>
                Bottom
                <Select
                    optionsArr={['cowboy-bottom', 'jean-bottom', 'diaper-bottom', 'skirt-bottom', 'suit-bottom']}
                    setState={setBottom}
                    state={bottom}
                />
            </label>
            <label>
                Catchphrase
                <input type="text" value={newCatchPhraseStr} onChange={(e) => setNewCatchphraseStr(e.target.value)} />
            </label>
            <button onClick={() => addCatchphrase()}>Add Catchphrase</button>
        </section>
    )
}
