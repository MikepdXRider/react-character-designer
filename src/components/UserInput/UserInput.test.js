import {screen, render} from '@testing-library/react'
import UserInput from './UserInput.jsx'

it('renders UserInput component', () => {
    // render enables us to pseudo-render a component. This also lets us use the screen method later on
    const {container} = render(<UserInput optionsArr={['test1', 'test2', 'test3']} />);
    // getByLabelText allows us to get the text within a <label> jsx tag.
    const headLabel = screen.getByLabelText('Head');
    const torsoLabel = screen.getByLabelText('Torso');
    const bottomLabel = screen.getByLabelText('Bottom');
    // expect the above elements to be in the rendered document.
    expect(headLabel).toBeInTheDocument();
    expect(torsoLabel).toBeInTheDocument();
    expect(bottomLabel).toBeInTheDocument();
    // expect container to match the produced snapshot(first test generates, all other test compare)
    expect(container).toMatchSnapshot();
})