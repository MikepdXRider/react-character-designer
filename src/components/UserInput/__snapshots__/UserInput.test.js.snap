// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders UserInput component 1`] = `
<div>
  <section>
    <label>
      Head
      <select>
        <option
          disabled=""
          hidden=""
          value=""
        >
          Please Choose...
        </option>
        <option
          data-testid="cowboy-hat"
          value="cowboy-hat"
        >
          cowboy-hat
        </option>
        <option
          data-testid="green-hat"
          value="green-hat"
        >
          green-hat
        </option>
        <option
          data-testid="top-hat"
          value="top-hat"
        >
          top-hat
        </option>
        <option
          data-testid="wizard-hat"
          value="wizard-hat"
        >
          wizard-hat
        </option>
      </select>
    </label>
    <label>
      Torso
      <select>
        <option
          disabled=""
          hidden=""
          value=""
        >
          Please Choose...
        </option>
        <option
          data-testid="cowboy-torso"
          value="cowboy-torso"
        >
          cowboy-torso
        </option>
        <option
          data-testid="formal-torso"
          value="formal-torso"
        >
          formal-torso
        </option>
        <option
          data-testid="jacket-torso"
          value="jacket-torso"
        >
          jacket-torso
        </option>
      </select>
    </label>
    <label>
      Bottom
      <select>
        <option
          disabled=""
          hidden=""
          value=""
        >
          Please Choose...
        </option>
        <option
          data-testid="cowboy-bottom"
          value="cowboy-bottom"
        >
          cowboy-bottom
        </option>
        <option
          data-testid="jean-bottom"
          value="jean-bottom"
        >
          jean-bottom
        </option>
        <option
          data-testid="diaper-bottom"
          value="diaper-bottom"
        >
          diaper-bottom
        </option>
        <option
          data-testid="skirt-bottom"
          value="skirt-bottom"
        >
          skirt-bottom
        </option>
        <option
          data-testid="suit-bottom"
          value="suit-bottom"
        >
          suit-bottom
        </option>
      </select>
    </label>
    <label>
      Catchphrase
      <input
        type="text"
        value=""
      />
    </label>
    <button>
      Add Catchphrase
    </button>
  </section>
</div>
`;
